### ISSUE TITLE
Bodies of messages moved to error queue are not encrypted even if the original body was

### ISSUE THREAD

**Issue by BredStik** [rebus-org/Rebus#1179]
I have noticed that messages moved to the error queue have their bodies in plain text (base64) even if the original was encrypted.  It seems like the outgoing pipeline is ignored when the message is handled by the DeadletterQueueErrorHandler.  I would have expected them to be encrypted if the original message was also encrypted.  This leaves a potential security flaw where messages could be read when in the error queue.

I have concocted a workaround by decorating the IErrorHandler but it seems a bit hacky.

```
opt.Decorate<IErrorHandler>(ctx => new ErrorHandlerEncryptorDecorator(ctx.Get<IErrorHandler>(), ctx.Get<IAsyncEncryptor>()));
```

```
public class ErrorHandlerEncryptorDecorator: IErrorHandler
{
    private readonly IErrorHandler _innerHandler;
    private readonly IAsyncEncryptor _encryptor;

    public ErrorHandlerEncryptorDecorator(IErrorHandler innerHandler, IAsyncEncryptor encryptor)
    {
        _innerHandler = innerHandler;
        _encryptor = encryptor;
    }
    
    public async Task HandlePoisonMessage(TransportMessage transportMessage, ITransactionContext transactionContext,
        ExceptionInfo exception)
    {
        var headers = transportMessage.Headers.Clone();

        var encryptedData = await _encryptor.Encrypt(transportMessage.Body);
        
        headers[EncryptionHeaders.ContentEncryption] = _encryptor.ContentEncryptionValue;
        headers[EncryptionHeaders.ContentInitializationVector] = Convert.ToBase64String(encryptedData.Iv);

        if(!string.IsNullOrEmpty(encryptedData.KeyId))
            headers[EncryptionHeaders.KeyId] = encryptedData.KeyId;

        var encryptedMessage = new TransportMessage(headers, encryptedData.Bytes);
        
        await _innerHandler.HandlePoisonMessage(encryptedMessage, transactionContext, exception);
    }
}
```

I would gladly work on a pull request for a permanent fix if you give me pointers on where it should be implemented properly...

**Comment by mookid8000** [comment]
That's clearly a regression, as the `OriginalTransportMessage` was originally introduced to handle this particular issue.

Let me just take a look at see if I can solve it easily without messing up too many other things 😅

**Comment by mookid8000** [comment]
It's fixed in Rebus 8.6.1, which is on NuGet.org now! 🙂 
THanks for reporting this issue.

**Comment by BredStik** [comment]
Thanks a lot for the quick turnaround!
